screens/auth/RegisterScreen.tsx

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+import { StyleSheet, Text, View, TextInput, Button } from 'react-native';
+import { useForm, Controller } from 'react-hook-form';
+import firebase from '../../firebase/clientApp';
+import { Client } from '../../types/types';
+import { setClient } from '../../firebase/queries';
+
+const auth = firebase.auth();
+
+export const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    alignItems: 'center',
+    backgroundColor: 'white',
+    width: '100%',
+    height: '100%',
+    paddingHorizontal: '12%',
+  },
+  content: {
+    height: '80%',
+    width: '100%',
+    display: 'flex',
+    justifyContent: 'center',
+  },
+  input: {
+    marginVertical: 10,
+    border: '1px solid black',
+    borderRadius: 5,
+    padding: 5,
+  },
+  red: {
+    color: 'red',
+  },
+  button: {
+    width: '80%',
+  },
+});
+
+type RegisterData = {
+  name: string;
+  email: string;
+  password: string;
+  language: string; // TODO: edit
+};
+
+const RegisterScreen = ({ navigation }) => {
+  const { control, handleSubmit } = useForm();
+
+  const onRegister = async (data: RegisterData) => {
+    // TODO: error handling
+    await auth
+      .createUserWithEmailAndPassword(data.email, data.password)
+      .then(async res => {
+        res.user.updateProfile({ displayName: data.name });
```

Comment:
Error: `Object is possibly 'null'.ts(2531) (parameter) res: firebase.auth.UserCredential`. I think this is happening because you're trying to update a profile that might not exist yet. A lag happens after `createUserWithEmailAndPassword`, which might also being causing name to not render on first sign in. Instead of updating profiles, you should create a user first, then retrieve the users collection from Firebase and `set` their content. 

Here's a [good example: ](https://medium.com/swlh/expo-firebase-authentication-cloud-firestore-using-async-await-react-hooks-700920ad4b2b) cmd+f "Step 5: Create an API folder..." 
![Screen Shot 2021-11-03 at 12 22 10 AM](https://user-images.githubusercontent.com/63089231/140021969-454d4f90-38a1-4cb5-b66f-65e303f22ad0.png)

I also recommend separating your Firebase functions into another file like this example so that it's easier to error handle and catch when users might not exist yet.

Reply:
Code snippet explanation from medium article: 
- The registration function takes in four arguments from “SignUp.js”
createUserWithEmailAndPassword()requires two arguments, email, and password, to create an account for the user on the “Authentication” of your Firebase Console. It automatically assigns a unique UID to identify each user.
firebase.firestore().collection('users')create a collection ( a.k.a. tables or models) called “users”
- The .doc(uid)on line 12 create a document (a.k.a an entry or instance) that contains a unique ID(uid) from the “Authentication”
- The .set() on line 13 represents the customized field (i.e., column name or attribute) for the collection. It can be any information that you wish to collect from your user. Note that this method takes in the field and its value as a key-value pair.
- In this example, we collected the email, first name, and last name of the user.